newrelic/resource_newrelic_cloud_azure_link_account.go

```diff
@@ -0,0 +1,227 @@
+				Required:    true,
+				ForceNew:    false,
+			},
+			"subscription_id": {
+				Type:        schema.TypeString,
+				Description: "subscription id for the Azure account",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"tenant_id": {
+				Type:        schema.TypeString,
+				Description: "tenant id for the Azure account",
+				Required:    true,
+				ForceNew:    true,
+			},
+		},
+	}
+}
+
+func resourceNewRelicCloudAzureLinkAccountCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConfig := meta.(*ProviderConfig)
+	client := providerConfig.NewClient
+	accountID := selectAccountID(providerConfig, d)
+	linkAccountInput := expandAzureCloudLinkAccountInput(d)
+	var diags diag.Diagnostics
+
+	cloudLinkAccountPayload, err := client.Cloud.CloudLinkAccountWithContext(ctx, accountID, linkAccountInput)
+
+	if err != nil {
+		diag.FromErr(err)
+	}
+
+	if len(cloudLinkAccountPayload.Errors) > 0 {
+		for _, err := range cloudLinkAccountPayload.Errors {
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  err.Type + " " + err.Message,
+			})
+		}
+	}
+
+	d.SetId(strconv.Itoa(cloudLinkAccountPayload.LinkedAccounts[0].ID))
+
+	if len(diags) > 0 {
+		return diags
+	}
+	return nil
+}
+
+// Extracting the Azure credentials from Schema using expandAzureCloudLinkAccountInput
+
+func expandAzureCloudLinkAccountInput(d *schema.ResourceData) cloud.CloudLinkCloudAccountsInput {
+
+	azureAccount := cloud.CloudAzureLinkAccountInput{}
+
+	if applicationID, ok := d.GetOk("application_id"); ok {
+		azureAccount.ApplicationID = applicationID.(string)
+	}
+
+	if clientSecretID, ok := d.GetOk("client_secret_id"); ok {
+		azureAccount.ClientSecret = cloud.SecureValue(clientSecretID.(string))
+	}
+
+	if name, ok := d.GetOk("name"); ok {
+		azureAccount.Name = name.(string)
+	}
+
+	if subscriptionID, ok := d.GetOk("subscription_id"); ok {
+		azureAccount.SubscriptionId = subscriptionID.(string)
+	}
+
+	if tenantID, ok := d.GetOk("tenant_id"); ok {
+		azureAccount.TenantId = tenantID.(string)
+	}
+
+	input := cloud.CloudLinkCloudAccountsInput{
+		Azure: []cloud.CloudAzureLinkAccountInput{azureAccount},
+	}
+
+	return input
+}
+
+func resourceNewRelicCloudAzureLinkAccountRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConfig := meta.(*ProviderConfig)
+	client := providerConfig.NewClient
+	accountID := selectAccountID(providerConfig, d)
+	linkedAccountID, convErr := strconv.Atoi(d.Id())
+
+	if convErr != nil {
+		return diag.FromErr(convErr)
+	}
+
+	linkedAccount, err := client.Cloud.GetLinkedAccount(accountID, linkedAccountID)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	readAzureLinkedAccount(d, linkedAccount)
+
+	return nil
+}
+
+//readAzureLinkedAccount function to read the outputs.
+
+func readAzureLinkedAccount(d *schema.ResourceData, result *cloud.CloudLinkedAccount) {
+	_ = d.Set("account_id", result.ExternalId)
```

Comment:
This should set to the nrAccountId
```suggestion
	_ = d.Set("account_id", result.nrAccountId)
```